pkg/streamingpromql/operators/binary_operation_test.go

```diff
@@ -208,6 +240,30 @@ func TestBinaryOperation_SeriesMerging(t *testing.T) {
 			},
 		},
 		"input series with conflict": {
```

Comment:
It'd be good to rename the test cases above to make it clear that they exercise the "floats only" case.